python/examples/miscellaneous/custom_action_demo/direct_function_call.py

```diff
@@ -0,0 +1,48 @@
+import typing as t
+from composio_openai import ComposioToolSet, action, Action
+from openai import OpenAI
+import base64
+
+from dotenv import load_dotenv
+load_dotenv()
+
+openai_client = OpenAI()
+toolset = ComposioToolSet(entity_id="abhishek")
+
+
+@action(toolname="gmail")
+def create_draft(
+    thread_id: str,
+    message_body: str,
+    execute_request: t.Callable,
```

Comment:
The type hint for `execute_request` parameter should be more specific. Consider using:
```python
execute_request: t.Callable[[str, str, dict, t.Optional[dict]], dict]
```
This better documents the expected signature of the function.